EIPS/eip-5749.md

```diff
@@ -0,0 +1,111 @@
+---
+eip: 5749
+title: Deprecate 'window.ethereum'
+description: Add 'window.evmproviders' and suggest the eventual removal of 'window.ethereum'
+author: Kosala Hemachandra (@kvhnuke), Brett Kolodny (@brettkolodny)
+discussions-to: https://ethereum-magicians.org/t/eip-5749-deprecate-window-ethereum/11195
+status: Draft
+type: Standards Track
+category: Interface
+created: 2022-10-04
+requires: 1193
+---
+
+
+## Abstract
+
+A Javascript Ethereum Provider interface injection for interoperability between multiple wallets at the same time.
+
+## Motivation
+
+At present, `window.ethereum` is the prevailing method by which Ethereum-compatible applications interact with injected wallets. This originated with Mist Wallet in 2015 to interact with other applications. With the proliferation of both applications and wallets, `window.ethereum` has unintended negative consequences:
+
+- `window.ethereum` only permits one wallet to be injected at a time, resulting in a race condition between two or more wallets. This creates an inconsistent connection behavior that makes having and using more than one browser wallet unpredictable and impractical. The current solution is for wallets to inject their own namespaces, but this is not feasible as every application would need to be made aware of any wallet that might be used.
+- The aforementioned race condition means users are disincentivized to experiment with new wallets. This creates a 'winner-takes-most' wallet market across EVM chains which forces application developers to optimize for a particular wallet experience.
+- The 'winner-takes-most' wallet environment that results from the `window.ethereum` standard hinders innovation because it creates a barrier to adoption. New entrants into the space have difficulty gaining traction against legacy players because users can have no more than one injected wallet. With new entrants crowded out, legacy wallet providers are put under little pressure to innovate.
+- Wallets continue to be the most fundamental tool for interacting with blockchains. A homogeneous wallet experience in Ethereum and EVM chains risks stunting UX improvement across the ecosystem and will allow other ecosystems that are more encouraging of competition and innovation to move ahead. 
+
+Replacing `window.ethereum` with `window.evmproviders` will allow solutions such as web3modal and web3onboard to only display all injected wallets the user has installed. This will simpify the UX and remove race conditions between wallet providers in case multiple wallets are installed. Over time, as `window.evmproviders` supplants the current standard and removes barriers to choice, we can hope to see a wallet landscape more reflective of user preference.
+
+### Some wallets that currently use `window.ethereum`
+	Metamask
+	Coinbase wallet
+	Enkrypt
+	Rabby
+	Trust wallet
+	Rainbow ..etc
+
```

Comment:
There is no way this list stands the test of time, and it doesn't add value to this standard so I recommend removing it.  EIPs are not the right place to store lists of things.

Reply:
IMHO, I am in favor of having this list with "as of 2022-10...". It's a good part of the motivation argument

Reply:
What does this add to the motivation argument? It's not clear right now. Please either remove the list or explain why the list is useful **in the motivation section**.

Reply:
let me move it to motivation section, and reference this month. I think itll make it clear why we needed this eip in the first place

Reply:
moved to motivation

Reply:
Would it serve the same purpose to just say "many wallets use `window.ethereum`?

Reply:
> Would it serve the same purpose to just say "many wallets use `window.ethereum`"?

IMHO - yes.